client/object_search.go

```diff
@@ -175,6 +175,27 @@ func (x *ObjectListReader) Read(buf []oid.ID) (int, bool) {
 	}
 }
 
+// Iterate iterates over the list of found object identifiers.
+// Returns an error if object can't be read.
+//
+// Panics if buf has zero length.
+func (x *ObjectListReader) Iterate(f func(oid.ID) bool) error {
+	buf := make([]oid.ID, 1)
+
+	for {
+		// Do not check first return value because `len(buf) == 1`,
+		// so false means nothing was read.
+		_, ok := x.Read(buf)
+		if !ok {
+			_, err := x.Close()
```

Comment:
We ignore `ResObjectSearch` because `Iterate` is expected to be used in simple `Client` mode only (`WithNeoFSErrorParsing` option)? If so, lets mention it in docs, otherwise we need extra condition:
```
if err == nil {
  err = apistatus.ErrFromStatus(res.Status())
}
```